pkg/virt-controller/network/ipamclaims_test.go

```diff
@@ -0,0 +1,199 @@
+)
+
+const (
+	redNetworkName     = "red"
+	redNetworkSrcName  = redNetworkName + "-net"
+	blueNetworkName    = "blue"
+	blueNetworkSrcName = blueNetworkName + "-net"
+	namespace          = "test-ns"
+	vmiName            = "testvmi"
+	vmiUID             = vmiName + "UID"
+	networkName        = "network_name"
+)
+
+var _ = Describe("CreateIPAMClaims", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	var networkClient *fakenetworkclient.Clientset
+	var vmi *virtv1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		networkClient = fakenetworkclient.NewSimpleClientset()
+		virtClient.EXPECT().NetworkClient().Return(networkClient).AnyTimes()
+		ipamClaimsClient := fakeipamclaimclient.NewSimpleClientset()
+		virtClient.EXPECT().IPAMClaimsClient().Return(ipamClaimsClient).AnyTimes()
+	})
+
+	BeforeEach(func() {
+		vmi = api.NewMinimalVMI(vmiName)
+		vmi.Namespace = namespace
+		vmi.Spec.Domain.Devices.Interfaces = []virtv1.Interface{
+			{Name: redNetworkName},
+			{Name: "absent", State: virtv1.InterfaceStateAbsent},
+		}
+		vmi.UID = vmiUID
+		vmi.Spec.Networks = []virtv1.Network{
+			{
+				Name:          redNetworkName,
+				NetworkSource: virtv1.NetworkSource{Multus: &virtv1.MultusNetwork{NetworkName: redNetworkSrcName}},
+			},
+			{
+				Name:          "absent",
+				NetworkSource: virtv1.NetworkSource{Multus: &virtv1.MultusNetwork{NetworkName: "absent-net"}},
+			},
+		}
+	})
+
+	Context("With allowPersistentIPs enabled in the NADs", func() {
+		BeforeEach(func() {
+			persistentIPs := map[string]bool{redNetworkSrcName: true}
+			err := createNADs(networkClient, namespace, vmi.Spec.Networks, persistentIPs)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should create the expected IPAMClaims", func() {
+			err := CreateIPAMClaims(virtClient, vmi, nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			ipamClaimsList, err := virtClient.IPAMClaimsClient().K8sV1alpha1().IPAMClaims(namespace).List(
+				context.Background(),
+				v1.ListOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaimsList.Items).To(HaveLen(1))
+			Expect(ipamClaimsList.Items[0].OwnerReferences).To(HaveLen(1))
+
+			claim := ipamClaimsList.Items[0]
+			Expect(claim.OwnerReferences[0]).To(Equal(v1.OwnerReference{
+				APIVersion:         "kubevirt.io/v1",
+				Kind:               "VirtualMachineInstance",
+				Name:               vmiName,
+				UID:                vmiUID,
+				Controller:         pointer.P(true),
+				BlockOwnerDeletion: pointer.P(true),
+			}))
+			Expect(claim.Name).To(Equal(fmt.Sprintf("%s.%s", vmiName, redNetworkName)))
+			Expect(claim.Namespace).To(Equal(namespace))
+			Expect(claim.Spec).To(Equal(ipamclaims.IPAMClaimSpec{
+				Network:   networkName,
+				Interface: namescheme.GenerateHashedInterfaceName(redNetworkName),
```

Comment:
This is like duplicate the code, can we just put the literal claimName and interface ? similar to what you are doing at the annotation tests, it feels redundant but it will really test what we need.

Reply:
Done, but imo at least the Network should use the const `nadNetworkName` (atm hardcoded as you asked)
note please it is nad's spec.config.name here, not ipamClaim (which is right only for the annotation)